tests/test_flags.py

```diff
@@ -0,0 +1,294 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Type
+
+import pytest
+
+from disnake import flags
+
+if TYPE_CHECKING:
+
+    class TestFlags(flags.BaseFlags):
+        """A test class for flag testing."""
+
+        @flags.flag_value
+        def one(self):
+            return 1 << 0
+
+        @flags.flag_value
+        def two(self):
+            return 1 << 1
+
+        @flags.flag_value
+        def four(self):
+            return 1 << 2
+
+
+def test_flag_value_creation() -> None:
+    flag = flags.flag_value(lambda x: 1 << 2)
+    assert 1 << 2 == flag.flag
+
+
+def test_fill_with_flags() -> None:
+    @flags.fill_with_flags()
+    class TestFlags(flags.BaseFlags):
+        """A test class for flag testing."""
+
+        @flags.flag_value
+        def one(self):
+            return 1 << 0
+
+        @flags.flag_value
+        def two(self):
+            return 1 << 1
+
+        @flags.flag_value
+        def four(self):
+            return 1 << 2
+
+    ins = TestFlags()
+
+    assert ins
+
+
+@pytest.fixture()
+def test_flags() -> Type[flags.BaseFlags]:
```

Comment:
There should also be tests that make use of `alias_flag_value`

Reply:
Added a few, should add more

Reply:
Looks good, could also test getting/setting that specific alias flag.

Reply:
https://github.com/DisnakeDev/disnake/pull/616/commits/eaded3b3fcbaf9bdc675c7674a0a2acaf58dc163